services/src/main/java/org/exoplatform/task/integration/TaskCommentAttachmentPlugin.java

```diff
@@ -0,0 +1,113 @@
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.exoplatform.task.integration;
+
+import org.exoplatform.commons.exception.ObjectNotFoundException;
+import org.exoplatform.services.security.Identity;
+import org.exoplatform.social.attachment.AttachmentPlugin;
+import org.exoplatform.social.core.space.model.Space;
+import org.exoplatform.social.core.space.spi.SpaceService;
+import org.exoplatform.task.dto.CommentDto;
+import org.exoplatform.task.service.ProjectService;
+import org.exoplatform.task.service.TaskService;
+import org.exoplatform.task.service.CommentService;
+import org.exoplatform.task.util.TaskUtil;
+
+
+public class TaskCommentAttachmentPlugin extends AttachmentPlugin {
+
+  public static final String TASK_COMMENT_OBJECT_TYPE                 = "taskComment";
+
+  public static final String TASK_COMMENT_NOT_FOUND_EXCEPTION_MESSAGE = "Task comment with id %s wasn't found";
+
+  private TaskService        taskService;
+
+  private CommentService     commentService;
+
+  private ProjectService     projectService;
+
+  private SpaceService       spaceService;
+
+  public TaskCommentAttachmentPlugin(TaskService taskService,
+                                     CommentService commentService,
+                                     ProjectService projectService,
+                                     SpaceService spaceService) {
+    this.taskService = taskService;
+    this.commentService = commentService;
+    this.projectService = projectService;
+    this.spaceService = spaceService;
+  }
+
+  @Override
+  public String getObjectType() {
+    return TASK_COMMENT_OBJECT_TYPE;
+  }
+
+  @Override
+  public boolean hasAccessPermission(Identity userIdentity, String entityId) throws ObjectNotFoundException {
+    try {
+      CommentDto comment = commentService.getComment(Long.parseLong(entityId));
+      return TaskUtil.hasViewPermission(taskService, comment.getTask(), userIdentity);
+    } catch (Exception e) {
+      throw new ObjectNotFoundException(String.format(TASK_COMMENT_NOT_FOUND_EXCEPTION_MESSAGE, entityId));
+    }
+  }
+
+  @Override
+  public boolean hasEditPermission(Identity userIdentity, String entityId) throws ObjectNotFoundException {
+    try {
+      CommentDto comment = commentService.getComment(Long.parseLong(entityId));
+      return TaskUtil.hasEditPermission(taskService, comment.getTask(), userIdentity);
```

Comment:
On Comment creator should be able to change the attached images to a comment through REST APIs. @srenault-meeds to be compliant with other features, ins't it ?

Reply:
Indeed, that mustn't be possible. I have added a dedicated task for it to explain expected behaviour

Reply:
ok, thanks, I think it would be better to wait for this merge until it's worked.

Reply:
ok task added here: https://builders.meeds.io/portal/meeds/tasks/taskDetail/2438
cc @SaraBoutej

Reply:
```suggestion
      return StringUtils.equals(comment.getAuthor(), userIdentity.getUserId());
```